subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -113,9 +118,45 @@ static const char *fmt_name(enum coap_content_format fmt)
 }
 #endif /* CONFIG_NRF_CLOUD_COAP_LOG_LEVEL_DBG */
 
+static struct cc_xfer_data *xfer_ctx_take(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(xfer_ctx_pool); i++) {
+		if (!atomic_test_and_set_bit(&xfer_ctx_pool[i].used, 0)) {
+			return &xfer_ctx_pool[i];
+		}
+	}
+	return NULL;
+}
+
+static void xfer_ctx_free(struct cc_xfer_data *ctx)
```

Comment:
```suggestion
static void xfer_ctx_release(struct cc_xfer_data *ctx)
```

just to avoid alloc/free wording.